Mathlib/Data/Matrix/PEquiv.lean

```diff
@@ -178,4 +178,28 @@ theorem toMatrix_toPEquiv_eq [DecidableEq n] [Zero α] [One α] (σ : Equiv.Perm
 
 @[deprecated (since := "2025-01-27")] alias equiv_toPEquiv_toMatrix := toMatrix_toPEquiv_eq
 
+@[simp]
+lemma map_toMatrix {β : Type*} [DecidableEq n] [NonAssocSemiring α] [NonAssocSemiring β]
+    (f : α →+* β) (σ : m ≃. n) : σ.toMatrix.map f = σ.toMatrix := by
+  ext i j
+  simp
+
 end PEquiv
+
+namespace Equiv
+
+open Matrix
+
+variable {α m n : Type*} [DecidableEq m] [DecidableEq n]
+
+lemma toMatrix_toPEquiv_mulVec_single [Fintype n] [NonAssocSemiring α] (σ : m ≃ n) (i : n) (x : α) :
+    σ.toPEquiv.toMatrix *ᵥ Pi.single i x = Pi.single (σ.symm i) x := by
+  ext j
+  simp [Pi.single_apply, Equiv.apply_eq_iff_eq_symm_apply σ]
+
+lemma single_vecMul_toMatrix_toPEquiv [Fintype m] [NonAssocSemiring α] (σ : m ≃ n) (i : m) (x : α) :
+    Pi.single i x ᵥ* σ.toPEquiv.toMatrix = Pi.single (σ i) x := by
+  ext j
+  simp [Pi.single_apply]
```

Comment:
Based on the lemmas above, I think these should remain in the `PEquiv` namespace, as they are about `PEquiv.toMatrix`